tests/integ/test_pandas_to_df.py

```diff
@@ -48,6 +48,80 @@ def tmp_table_complex(session):
         Utils.drop_table(session, table_name)
 
 
+@pytest.mark.parametrize("quote_identifiers", [True, False])
+@pytest.mark.parametrize("auto_create_table", [True, False])
+def test_write_pandas_with_overwrite(
+    session,
+    tmp_table_basic,
+    quote_identifiers: bool,
+    auto_create_table: bool,
+):
+    pd1 = PandasDF(
+        [
+            (1, 4.5, "Nike"),
+            (2, 7.5, "Adidas"),
+            (3, 10.5, "Puma"),
+        ],
+        columns=["id".upper(), "foot_size".upper(), "shoe_make".upper()],
+    )
+
+    pd2 = PandasDF(
+        [(1, 8.0, "Dia Dora")],
+        columns=["id".upper(), "foot_size".upper(), "shoe_make".upper()],
+    )
+
+    pd3 = PandasDF(
+        [(1, "dash", 1000, 32)],
+        columns=["id".upper(), "name".upper(), "points".upper(), "age".upper()],
+    )
+
+    table_name = tmp_table_basic
+
+    # Create initial table and insert 3 rows
+    drop_sql = f'DROP TABLE IF EXISTS "{table_name}"'
+    session.sql(drop_sql).collect()
+    df1 = session.write_pandas(
+        pd1, table_name, quote_identifiers=quote_identifiers, auto_create_table=True
+    )
+    results = df1.to_pandas()
+    assert_frame_equal(results, pd1, check_dtype=False)
+
+    # Insert 1 row
+    df2 = session.write_pandas(
+        pd2,
+        table_name,
+        quote_identifiers=quote_identifiers,
+        overwrite=True,
+        auto_create_table=auto_create_table,
+    )
+    results = df2.to_pandas()
+    # Results should match pd2
+    assert_frame_equal(results, pd2, check_dtype=False)
+
+    # Insert 1 row with new schema
+    if auto_create_table:
+        # In this case, the table is first dropped and since there's a new schema, the results should now match pd3
```

Comment:
what would happen if `auto_create_table` is `False` while `overwrite` is `True` for the new schema?

Reply:
Submitting a new PR with overwrite as a new pytest parameter.